binding/HarfBuzzSharp.Shared/Font.cs

```diff
@@ -163,54 +124,99 @@ public unsafe int[] GetVerticalGlyphAdvances (int count, IntPtr firstGlyph)
 			return advances;
 		}
 
-		public uint GetGlyph (int unicode) => GetGlyph (unicode, 0);
+		public bool TryGetHorizontalGlyphOrigin (uint glyph, out int xOrigin, out int yOrigin) =>
+			HarfBuzzApi.hb_font_get_glyph_h_origin (Handle, glyph, out xOrigin, out yOrigin);
+
+		public bool TryGetVerticalGlyphOrigin (uint glyph, out int xOrigin, out int yOrigin) =>
+			HarfBuzzApi.hb_font_get_glyph_v_origin (Handle, glyph, out xOrigin, out yOrigin);
+
+		public int GetHorizontalGlyphKerning (uint leftGlyph, uint rightGlyph) =>
+			HarfBuzzApi.hb_font_get_glyph_h_kerning (Handle, leftGlyph, rightGlyph);
+
+		public bool TryGetGlyphExtents (uint glyph, out GlyphExtents extents) =>
+			HarfBuzzApi.hb_font_get_glyph_extents (Handle, glyph, out extents);
+
+		public bool TryGetGlyphContourPoint (uint glyph, uint pointIndex, out int x, out int y) =>
+			HarfBuzzApi.hb_font_get_glyph_contour_point (Handle, glyph, pointIndex, out x, out y);
 
-		public uint GetGlyph (int unicode, int variationSelector = 0)
+		public unsafe bool TryGetGlyphName (uint glyph, out string name)
 		{
-			if (unicode < 0) {
-				throw new ArgumentOutOfRangeException (nameof (unicode), "Unicode must be non negative.");
+			var buffer = ArrayPool<char>.Shared.Rent (NameBufferLength);
+			try {
+				fixed (char* first = buffer) {
+					if (!HarfBuzzApi.hb_font_get_glyph_name (Handle, glyph, first, buffer.Length)) {
+						name = string.Empty;
+						return false;
+					}
+					name = new string (first);
```

Comment:
Instead of checking for a null character, let the string constructor do the work.